src/main/java/ch/njol/skript/expressions/ExprStructureInfo.java

```diff
@@ -0,0 +1,156 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.bukkit.Bukkit;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.BlockState;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.structure.Structure;
+import org.bukkit.util.BlockVector;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.BlockStateBlock;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.iterator.ArrayIterator;
+
+@Name("Structure Info")
+@Description("An expression to obtain information about a structure's entities, size, and blocks.")
+@Examples({
+	"loop all entities of structure {_structure}:",
+		"\tif loop-entity is a diamond:",
+			"\t\tmessage \"Race to the diamond at %loop-entity's location%\" to {_players::*}",
+			"\t\tstop",
+	"if the length of {_structure}'s vector is greater than 100:",
+		"\tmessage \"&a+50 coins will be granted for winning on this larger map!\""
+})
+@RequiredPlugins("Minecraft 1.17.1+")
+@Since("INSERT VERSION")
+public class ExprStructureInfo extends PropertyExpression<Structure, Object> {
+
+	static {
+		if (Skript.classExists("org.bukkit.structure.Structure"))
+			register(ExprStructureInfo.class, Object.class, "[structure] (:blocks|:entities|name:name[s]|size:(size|[lowest] [block] vector)[s])", "structures");
```

Comment:
Causing conflicts with vector from location
``vector of %structure%``

Reply:
How is this conflicting? There is no object in the pattern?

Reply:
this pattern can be `vector of %structure%`, which conflicts with `vector of %location%`

Reply:
Caused by not having https://github.com/SkriptLang/Skript/pull/5457

Reply:
ok but that doesn't mean you can ignore it